### PR TITLE
abstracted out the model for #24 and #3

### PR DESCRIPTION
**Pull request by Josh-Joseph**

### CODE REVIEW

File: configs/train-large.yaml [hunk 1]
```diff
@@ -1,7 +1,15 @@
 train_config:
   model_id_prefix: 'segmentation-model-large'
   dataset_id: 'dataset-large'
+  segmentation_model:
```
Comment (Josh-Joseph) [inline configs/train-large.yaml]: @rak5216 check out the configs especially - make sure they jive with you

Comment (rak5216) [reply configs/train-large.yaml]: looks great. do you want me to verify the changes work, or perhaps you've likely done that?

Comment (Josh-Joseph) [reply configs/train-large.yaml]: I have but for decide sized changes like these (that touch many different files) it'd probably be worthwhile for you to test it too. I'm not sure we've talked about how to change/test a different branch -- if not, happy to talk through it tomorrow

File: models.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from keras.optimizers import Adam
+from keras.metrics import accuracy, binary_crossentropy, categorical_crossentropy
+from segmentation_models import Unet
+from segmentation_models.metrics import iou_score
+from segmentation_models.losses import jaccard_loss, dice_loss
+
```
Comment (rak5216) [inline models.py]: So this file is entirely new? Where does `generate_compiled_segmentation_model` show up since you're importing it in `train_seg_model.py`?

File: image_utils.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def on_epoch_end(self, epoch, logs=None):
 
 # adapted from: https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly
 class ImagesAndMasksGenerator(Sequence):
-    def __init__(self, dataset_directory, rescale, target_size, batch_size, shuffle=True, seed=None, random_rotation=False):
+    def __init__(self, dataset_directory, rescale, target_size, batch_size, shuffle=False, seed=None, random_rotation=False):
```
Comment (rak5216) [inline image_utils.py]: Should `training_data_shuffle_seed` from config be passed to here?

File: models.py (hunk 2 above)
Comment (Josh-Joseph) [reply models.py]: yep it's new -- the idea is that previously unet (or whatever model/parameters we wanted to use) was hard coded into train, test, and infer. this makes it a real pain to try out different parameters and architectures. by introducing models.py we can make any model changes there (or really in the config file) and then have those used in train, test, and infer without having to touch their code

File: image_utils.py (hunk 3 above)
Comment (rak5216) [reply image_utils.py]: nvm i guess, since I see the seed below in `train_segmentation_model.py`

File: train_segmentation_model.py [hunk 4]
```diff
@@ -77,26 +73,22 @@ def train(gcp_bucket, config_file):
         target_size=target_size,
         batch_size=batch_size,
         shuffle=True,
-        random_rotation=train_config['augmentation']['random_90-degree_rotations'],
-        seed=None)
+        random_rotation=train_config['data_augmentation']['random_90-degree_rotations'],
+        seed=train_config['training_data_shuffle_seed'])
```
Comment (rak5216) [inline train_segmentation_model.py]: seed noted

File: image_utils.py (hunk 3 above)
Comment (Josh-Joseph) [reply image_utils.py]: yep - it does. in line 70 in train_segmentation_model.py you'll see the training generator is constructed by:
```
    train_generator = ImagesAndMasksGenerator(
        Path(local_dataset_dir, train_config['dataset_id'], 'train').as_posix(),
        rescale=1./255,
        target_size=target_size,
        batch_size=batch_size,
        shuffle=True,
        random_rotation=train_config['data_augmentation']['random_90-degree_rotations'],
        seed=train_config['training_data_shuffle_seed'])
```

File: configs/train-large.yaml (hunk 1 above)
Comment (rak5216) [reply configs/train-large.yaml]: so you mean test by cloning your branch before merging to master?

File: train_segmentation_model.py (hunk 4 above)
Comment (Josh-Joseph) [reply train_segmentation_model.py]: yeah exactly -- or maybe I'm not quite understanding your comment from above?

Comment (rak5216) [reply train_segmentation_model.py]: all good. my lack of python know how

File: configs/train-large.yaml (hunk 1 above)
Comment (rak5216) [reply configs/train-large.yaml]: everything looks great, should I clone  your branch?

Comment (rak5216) [reply configs/train-large.yaml]: new changes have been verified to work on my end @Josh-Joseph

**Comment by Josh-Joseph** [comment]
rad looks good

**Comment by Josh-Joseph** [comment]
This PR also addresses issue #26
